apps/common/utils/format.tsx

```diff
@@ -0,0 +1,228 @@
+import {type ReactElement, useMemo} from 'react';
+import {formatToNormalizedValue, toBigInt} from '@yearn-finance/web-lib/utils/format.bigNumber';
+import {isZero} from '@yearn-finance/web-lib/utils/isZero';
+
+type TAmountOptions = {
+	minimumFractionDigits?: number,
+	maximumFractionDigits?: number,
+	displayDigits?: number,
+	shouldDisplaySymbol?: boolean
+	shouldCompactValue?: boolean
+}
+type TAmount = {
+	value: bigint | number
+	decimals: number | bigint
+	symbol?: string
+	options?: TAmountOptions
+}
+const defaultOptions: TAmountOptions = {
+	minimumFractionDigits: 2,
+	maximumFractionDigits: 2,
+	displayDigits: 0,
+	shouldDisplaySymbol: true,
+	shouldCompactValue: true
+};
+function assertValidNumber(value: number | undefined, defaultValue: number, label: string): number {
+	if (value === undefined) {
+		return defaultValue;
+	}
+	if (value < 0) {
+		console.warn(`formatAmount: ${label} should be positive.`);
+		return (defaultValue);
```

Comment:
can we remove these parentheses from all the returns, not a thing in TS :)
```suggestion
		return defaultValue;
```